url47fcfc0133_baliosian_tau_fst_tgz-pJ8-uy_edu_fing_mina_fsa_test_old_TestComposition1J8/src/uy/edu/fing/mina/fsa/tffst/Transition.java

```diff
@@ -0,0 +1,266 @@
+/* 
+ * based on 
+ *
+ * dk.brics.automaton
+ * Copyright (C) 2001-2004 Anders Moeller
+ * All rights reserved.
+ *
+ * es.upc.tffst
+ * Copyright (C) 2004 Javier Baliosian
+ * All rights reserved.
+ *
+ */
+
+package uy.edu.fing.mina.fsa.tffst;
+
+import java.io.Serializable;
+
+import uy.edu.fing.mina.fsa.tf.TfI;
+import uy.edu.fing.mina.fsa.tf.TfString;
+
+
+public class Transition implements Serializable {
+
+  static final long serialVersionUID = 40001;
+
+  TfString labelIn;
+
+  TfString labelOut;
+
+  private State to = null;
+
+  private State from = null;
+
+  /**
+   * @return the from
+   */
+  public State getFrom() {
+    return from;
+  }
+
+  /**
+   * 
+   * @param from the from to set
+   */
+  public void setFrom(State from) {
+    if (this.from != from) {
+      State toBak = to;
+      if (this.to != null)   this.to.remInTran(this);
+      if (this.from != null) this.from.remOutTran(this);
+      this.from = from;
+      this.to = toBak;
+      if (this.from != null) this.from.addOutTran(this);
+      if (this.to != null) this.to.addInTran(this);
+    }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 46:54, 62:70.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.